data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3ObjectHandler.java

```diff
@@ -28,5 +28,24 @@ void parseS3Object(final S3ObjectReference s3ObjectReference,
                        final SourceCoordinator<S3SourceProgressState> sourceCoordinator,
                        final String partitionKey) throws IOException;
 
+    /**
+     * delete S3 object using S3 object reference
+     * @param s3ObjectReference Contains bucket and s3 object details
+     */
     void deleteS3Object(final S3ObjectReference s3ObjectReference);
+
+    /**
+     * process S3 object metadata using S3 object reference and pushing to buffer
+     * @param s3ObjectReference Contains bucket and s3 object details
+     * @param acknowledgementSet acknowledgement set for the object
+     * @param sourceCoordinator source coordinator
+     * @param partitionKey partition key
+     *
+     * @throws IOException exception is thrown every time because this is not supported
+     */
+    default void processS3ObjectMetadata(final S3ObjectReference s3ObjectReference,
```

Comment:
Rather than adding another method and having more conditionals in all the code, we can simplify this by implementing a new `S3ObjectHandler` which handles metadata-only.

Even better, I think we could have a different abstraction _within_ the `S3ObjectWorker` to handle this differently. This would help us share metric reporting.